### ISSUE TITLE
Unable to upload to Firebase Storage. 

### ISSUE THREAD

**Issue by sree-ksl** [androidthings/doorbell#18]
Having this issue: "Local module descriptor class for com.google.android.gms.firebasestorage not found." Using Pico iMX7 with android things dev preview 7.

My Logcat:
03-26 14:48:28.969 1652-1679/com.example.androidthings.doorbell I/DynamiteModule: Considering local module com.google.android.gms.firebase_database:4 and remote module com.google.android.gms.firebase_database:0
03-26 14:48:28.970 1652-1679/com.example.androidthings.doorbell I/DynamiteModule: Selected local version of com.google.android.gms.firebase_database
03-26 14:48:29.159 1652-1692/com.example.androidthings.doorbell W/DynamiteModule: Local module descriptor class for com.google.android.gms.firebasestorage not found.
03-26 14:48:29.160 1652-1692/com.example.androidthings.doorbell I/DynamiteModule: Considering local module com.google.android.gms.firebasestorage:0 and remote module com.google.android.gms.firebasestorage:0
03-26 14:48:29.168 1652-1692/com.example.androidthings.doorbell E/NetworkRqFactoryProxy: NetworkRequestFactoryProxy failed with a RemoteException:
                                                                                         com.google.android.gms.dynamite.DynamiteModule$zzc: No acceptable module found. Local version is 0 and remote version is 0.
                                                                                             at com.google.android.gms.dynamite.DynamiteModule.zza(Unknown Source:176)
                                                                                             at com.google.android.gms.internal.zzewy.<init>(Unknown Source:18)
                                                                                             at com.google.android.gms.internal.zzewy.zzh(Unknown Source:13)
                                                                                             at com.google.firebase.storage.StorageReference.zzcjq(Unknown Source:8)
                                                                                             at com.google.firebase.storage.UploadTask.run(Unknown Source:105)
                                                                                             at com.google.firebase.storage.zzr.run(Unknown Source:2)
                                                                                             at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
                                                                                             at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
                                                                                             at java.lang.Thread.run(Thread.java:764)
03-26 14:48:29.171 1652-1692/com.example.androidthings.doorbell E/UploadTask: Unable to create a network request from metadata
                                                                              android.os.RemoteException
                                                                                  at com.google.android.gms.internal.zzewy.<init>(Unknown Source:85)
                                                                                  at com.google.android.gms.internal.zzewy.zzh(Unknown Source:13)
                                                                                  at com.google.firebase.storage.StorageReference.zzcjq(Unknown Source:8)
                                                                                  at com.google.firebase.storage.UploadTask.run(Unknown Source:105)
                                                                                  at com.google.firebase.storage.zzr.run(Unknown Source:2)
                                                                                  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
                                                                                  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
                                                                                  at java.lang.Thread.run(Thread.java:764)
03-26 14:48:29.171 1652-1692/com.example.androidthings.doorbell E/StorageException: StorageException has occurred.
                                                                                    An unknown error occurred, please check the HTTP result code and inner exception for server response.
                                                                                     Code: -13000 HttpResult: 0
03-26 14:48:29.174 1652-1692/com.example.androidthings.doorbell E/StorageException: null
                                                                                    android.os.RemoteException
                                                                                        at com.google.android.gms.internal.zzewy.<init>(Unknown Source:85)
                                                                                        at com.google.android.gms.internal.zzewy.zzh(Unknown Source:13)
                                                                                        at com.google.firebase.storage.StorageReference.zzcjq(Unknown Source:8)
                                                                                        at com.google.firebase.storage.UploadTask.run(Unknown Source:105)
                                                                                        at com.google.firebase.storage.zzr.run(Unknown Source:2)
                                                                                        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
                                                                                        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
                                                                                        at java.lang.Thread.run(Thread.java:764)
03-26 14:48:29.174 1652-1692/com.example.androidthings.doorbell E/StorageException: StorageException has occurred.
                                                                                    An unknown error occurred, please check the HTTP result code and inner exception for server response.
                                                                                     Code: -13000 HttpResult: 0
03-26 14:48:29.175 1652-1652/com.example.androidthings.doorbell W/DoorbellActivity: Unable to upload image to Firebase
03-26 14:48:29.176 1652-1692/com.example.androidthings.doorbell E/StorageException: null
                                                                                    android.os.RemoteException
                                                                                        at com.google.android.gms.internal.zzewy.<init>(Unknown Source:85)
                                                                                        at com.google.android.gms.internal.zzewy.zzh(Unknown Source:13)
                                                                                        at com.google.firebase.storage.StorageReference.zzcjq(Unknown Source:8)
                                                                                        at com.google.firebase.storage.UploadTask.run(Unknown Source:105)
                                                                                        at com.google.firebase.storage.zzr.run(Unknown Source:2)
                                                                                        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
                                                                                        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
                                                                                        at java.lang.Thread.run(Thread.java:764)
03-26 14:48:29.239 1652-1693/com.example.androidthings.doorbell D/NetworkSecurityConfig: No Network Security Config specified, using platform default

**Comment by Fleker** [comment]
Can you verify that your device is connected to the Internet? Also, can you check the security rules of your Firebase storage project?

**Comment by sree-ksl** [comment]
I flashed the latest Android things image on the device and also setup Wifi. I did change the security rules in storage to all access.

**Comment by Fleker** [comment]
Are you currently filtering by **Show only selected application**? If you go to **No Filters** are there any other Firebase-related stacktraces or errors?

**Comment by sree-ksl** [comment]
I checked with **No Filters**, Did find Firebase-related stack traces same as before. I reflashed the android things image again on the board and also setup wifi. Rechecked the code and API keys and calls again and executed the app. Fingers crossed :neutral_face: . It worked as a charm. Also my firebase console and cloud api console shows the logs & requests that were made since the first time i installed the app. :flushed: !
